types/ndarray/index.d.ts

```diff
@@ -6,60 +6,67 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.3
 
-declare function ndarray<T = number>(
-    data: ndarray.Data<T>,
+declare function ndarray<D extends ndarray.Data = ndarray.Data<number>>(
+    data: D,
     shape?: number[],
     stride?: number[],
     offset?: number,
-): ndarray.NdArray<T>;
+): ndarray.NdArray<D>;
 
 declare namespace ndarray {
-    interface NdArray<T = number> {
-        data: Data<T>;
+    interface NdArray<D extends Data = Data<number>> {
+        data: D;
         shape: number[];
         stride: number[];
         offset: number;
-        dtype: DataType;
+        dtype: DataType<D>;
         size: number;
         order: number[];
         dimension: number;
-        get(...args: number[]): T;
-        set(...args: number[]): T;
-        index(...args: number[]): T;
-        lo(...args: number[]): NdArray<T>;
-        hi(...args: number[]): NdArray<T>;
-        step(...args: number[]): NdArray<T>;
-        transpose(...args: number[]): NdArray<T>;
-        pick(...args: Array<number | null>): NdArray<T>;
-        reshape(...shapes: number[]): NdArray<T>;
```

Comment:
I removed `reshape` as I don't believe it exists any more: https://github.com/scijs/ndarray/blob/master/ndarray.js

Reply:
The version number of this package is consistent with the version number of `ndarray` in which that function was removed?

Reply:
I don't believe this method was ever part of `ndarray`. I traced back its addition in `@types/ndarray` to this commit by @taoqf:  https://github.com/DefinitelyTyped/DefinitelyTyped/pull/18343/files. `numjs` augments `ndarray` with this method: https://github.com/nicolaspanel/numjs/blob/master/src/ndarray.js#L241, so the method should be typed in `@types/numjs` instead.

Reply:
Done! Thanks for getting me to double check this. 💯